crypto/evp/keymgmt_lib.c

```diff
@@ -243,14 +246,14 @@ OP_CACHE_ELEM *evp_keymgmt_util_find_operation_cache(EVP_PKEY *pk,
      */
     for (i = 0; i < end; i++) {
         p = sk_OP_CACHE_ELEM_value(pk->operation_cache, i);
-        if (keymgmt == p->keymgmt)
+        if (keymgmt == p->keymgmt && selection == p->selection)
```

Comment:
We could use the cache entry even if selection is subset of p->selection.

Reply:
Yes, I thought about this but wanted to go for the simplest approach as a first pass.
If you think this is important I can come up with a check that allows subsets.

Reply:
IMO it is trivial enough change to do it in this PR. But please first add a testcase to the fake RSA provider tests.

Reply:
I will fix this now that I have tests

Reply:
done